src/contract_client.rs

```diff
@@ -192,13 +194,78 @@ impl<Type> ContractClient<Type> {
             amount,
             method,
             parameter,
-            energy: 1_000_000.into(),
+            energy: None,
         };
 
         let invoke_result = self.client.invoke_instance(bi, &context).await?.response;
         Ok(invoke_result)
     }
 
+    /// Dry run an update. If the dry run succeeds the return value is an object
+    /// that has a send method to send the transaction that was simulated during
+    /// the dry run.
+    pub async fn dry_run_update<P: contracts_common::Serial, E>(
+        &mut self,
+        entrypoint: &str,
+        amount: Amount,
+        sender: AccountAddress,
+        message: &P,
+    ) -> Result<ContractUpdateBuilder, E>
+    where
+        E: From<NewReceiveNameError>
+            + From<RejectReason>
+            + From<v2::QueryError>
+            + From<ExceedsParameterSize>, {
+        let message = OwnedParameter::from_serial(message)?;
+        self.dry_run_update_raw(entrypoint, amount, sender, message)
+            .await
+    }
+
+    /// Like [`dry_run_update`](Self::dry_run_update) but expects an already
+    /// formed parameter.
+    pub async fn dry_run_update_raw<E>(
+        &mut self,
+        entrypoint: &str,
+        amount: Amount,
+        sender: AccountAddress,
+        message: OwnedParameter,
+    ) -> Result<ContractUpdateBuilder, E>
+    where
+        E: From<NewReceiveNameError> + From<RejectReason> + From<v2::QueryError>, {
+        let contract_name = self.contract_name.as_contract_name().contract_name();
+        let receive_name = OwnedReceiveName::try_from(format!("{contract_name}.{entrypoint}"))?;
+
+        let payload = UpdateContractPayload {
+            amount,
+            address: self.address,
+            receive_name,
+            message,
+        };
+
+        let context = ContractContext::new_from_payload(sender, Energy::from(10_000_000), payload);
```

Comment:
Why not just 

```suggestion
        let context = ContractContext::new_from_payload(sender, None, payload);
```

?